generate and exchange curve security keys on the fly (#3409)
flux-framework/flux-core: doc/man1/flux-keygen.rst

SYNOPSIS
========

**flux** **keygen** *PATH*

DESCRIPTION
===========

> flux-keygen(1) generates a long-term CURVE certificate used to secure
> a Flux system instance's overlay network.

Reviewer: ```suggestion
flux-keygen(1) generates a long-term CURVE certificate used to secure
the overlay network of a Flux system instance.
```

Reviewer: oops meant to start a review, not post a single comment, D'oh.

> The flux overlay network implements cryptographic privacy and data integrity

Reviewer: For consistency of capitalization:

```suggestion
The Flux overlay network implements cryptographic privacy and data integrity
```

when data is sent over a network.  Point to point ZeroMQ TCP connections
are protected with the CURVE security mechanism built into ZeroMQ
version 4, based on curve25519 and a CurveCP-like protocol.

All brokers participating in the system instance must use the same
certificate.  The certificate is part of the bootstrap configuration.

Flux instances that bootstrap with PMI do not require a configured certificate.

> In that case, each broker self-generates a unique certificate in and the
> public keys are exchanged with PMI.

Reviewer: ```suggestion
In that case, each broker self-generates a unique certificate and the
public keys are exchanged with PMI.
```
